apps/rush-lib/src/logic/deploy/DeployManager.ts

```diff
@@ -133,85 +136,131 @@ export class DeployManager {
+    }
 
-      const originalPackageJson: IPackageJson = JsonFile.load(
-        path.join(packageJsonRealFolderPath, 'package.json')
-      );
+    deployState.foldersToCopy.add(packageJsonRealFolderPath);
 
-      // Transform packageJson using pnpmfile.js
-      const packageJson: IPackageJson = deployState.pnpmfileConfiguration.transform(originalPackageJson);
+    const originalPackageJson: IPackageJson = JsonFile.load(
+      path.join(packageJsonRealFolderPath, 'package.json')
+    );
 
-      // Union of keys from regular dependencies, peerDependencies, optionalDependencies
-      // (and possibly devDependencies if includeDevDependencies=true)
-      const allDependencyNames: Set<string> = new Set<string>();
+    const sourceFolderInfo: IFolderInfo | undefined = deployState.folderInfosByPath.get(
+      FileSystem.getRealPath(packageJsonFolderPath)
+    );
+
+    // Transform packageJson using pnpmfile.js
+    const packageJson: IPackageJson = deployState.pnpmfileConfiguration.transform(originalPackageJson);
 
-      // Just the keys from optionalDependencies and peerDependencies
-      const optionalDependencyNames: Set<string> = new Set<string>();
+    // Union of keys from regular dependencies, peerDependencies, optionalDependencies
+    // (and possibly devDependencies if includeDevDependencies=true)
+    const allDependencyNames: Set<string> = new Set<string>();
 
-      for (const name of Object.keys(packageJson.dependencies || {})) {
+    // Just the keys from optionalDependencies and peerDependencies
+    const optionalDependencyNames: Set<string> = new Set<string>();
+
+    for (const name of Object.keys(packageJson.dependencies || {})) {
+      allDependencyNames.add(name);
+    }
+    if (deployState.scenarioConfiguration.json.includeDevDependencies) {
+      for (const name of Object.keys(packageJson.devDependencies || {})) {
         allDependencyNames.add(name);
       }
-      if (deployState.scenarioConfiguration.json.includeDevDependencies) {
-        for (const name of Object.keys(packageJson.devDependencies || {})) {
-          allDependencyNames.add(name);
+    }
+    for (const name of Object.keys(packageJson.peerDependencies || {})) {
+      allDependencyNames.add(name);
+      optionalDependencyNames.add(name); // consider peers optional, since they are so frequently broken
+    }
+    for (const name of Object.keys(packageJson.optionalDependencies || {})) {
+      allDependencyNames.add(name);
+      optionalDependencyNames.add(name);
+    }
+
+    if (sourceFolderInfo && sourceFolderInfo.isRushProject) {
+      const projectSettings: IDeployScenarioProjectJson | undefined =
+        sourceFolderInfo && sourceFolderInfo.projectSettings;
+      const additionalDependenciesToInclude: string[] | undefined =
+        projectSettings && projectSettings.additionalDependenciesToInclude;
+      const dependenciesToExclude: string[] | undefined =
+        projectSettings && projectSettings.dependenciesToExclude;
```

Comment:
The typescript version for this app does not support nullish coalescing or optional chaining, hence all the `&&`. I am not sure how much of an undertaking it is to upgrade the TS version

Reply:
We can't upgrade until we first integrate `downlevel-dts` into our toolchain.  (See https://github.com/microsoft/TypeScript/issues/36207 for details.)

Reply:
I simplified the expression a bit also.